src/core/FxMixer.cpp

```diff
@@ -327,6 +327,19 @@ void FxMixer::deleteChannel( int index )
 	for( int i = index; i < m_fxChannels.size(); ++i )
 	{
 		validateChannelName( i, i + 1 );
+
+		// set correct channel index
+		m_fxChannels[i]->m_channelIndex = i;
```

Comment:
To not do **this** on a function called `validateChannelName`? Do you have a better idea?